packages/@aws-cdk/aws-logs/lib/log-group.ts

```diff
@@ -537,4 +537,12 @@ export interface MetricFilterOptions {
    * @default No metric emitted.
    */
   readonly defaultValue?: number;
+
+  /**
+   * The fields to use as dimensions for the metric. One metric filter can include as many as three dimensions.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-logs-metricfilter-metrictransformation.html#cfn-logs-metricfilter-metrictransformation-dimensions
+   * @default - No dimensions attached to metrics.
+   */
+  readonly dimensions?: Record<string, string>;
```

Comment:
```
interface MetricDimensionProperty {
  readonly key: string;
  readonly value: string;
}
```
and 
```
dimensions: MetricDimensionProperty[];
```
Setting this contract here means that we can remove some duplicate code that exists in other places.

Reply:
Okay cool. I'll take your lead on it, but I think the other way would more easily match how dimensions are defined in the other packages - e.g Cloudwatch Metric class - https://github.com/aws/aws-cdk/blob/cea1039e3664fdfa89c6f00cdaeb1a0185a12678/packages/%40aws-cdk/aws-cloudwatch/lib/metric.ts#L255

Reply:
Hmm. I don't typically like adding them like this, but I do actually agree with you that they should be consistent. Consistency is more important than my preference.

Reply:
Sorry to cause extra work here, btw.

Reply:
@TheRealAmazonKendra No problem, I updated the PR again